### DISCUSSION TITLE
Is there a method to get the DataType of a tag?

### DISCUSSION THREAD

**Discussion by Destination2Unknown** [dmroeder/pylogix#226] [Q&A] [answered]
Is there an easy way to retrieve the data type of a PLC tag, or is it necessary to iterate over the tag list to obtain the data type?

**Comment by dmroeder** [comment, marked as answer]
If it is a UDT, then you have to get the tag list.

If it is one of the base data types:

```python
import pylogix

with pylogix.PLC("192.168.1.10") as comm:

    ret = comm.Read("BaseDINT")

    print(comm.KnownTags["BaseDINT"][0])
```

**Reply by Destination2Unknown** [reply]
Thanks, again!
